core/trino-main/src/main/java/io/trino/sql/planner/optimizations/ReplicateJoinAndSemiJoinInDelete.java

```diff
@@ -41,6 +46,21 @@ public PlanNode optimize(PlanNode plan, Session session, TypeProvider types, Sym
     {
         private boolean isDeleteQuery;
 
+        @Override
+        public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
+        {
+            PlanNode leftSourceRewritten = context.rewrite(node.getLeft(), context.get());
+            PlanNode rightSourceRewritten = context.rewrite(node.getRight(), context.get());
+
+            JoinNode rewrittenNode = (JoinNode) node.replaceChildren(ImmutableList.of(leftSourceRewritten, rightSourceRewritten));
+            Type joinType = rewrittenNode.getType();
+            if (isDeleteQuery && (joinType == INNER || joinType == LEFT)) {
+                return rewrittenNode.withDistributionType(REPLICATED);
```

Comment:
This should be applied to Joins directly between TableScan and DeleteNode, not to all Joins in the plan